tests/fuzz_xpath.cpp

```diff
@@ -1,26 +1,37 @@
 #include "../src/pugixml.hpp"
+#include "fuzzer/FuzzedDataProvider.h"
 
 #include <stdint.h>
 #include <string.h>
+#include <string>
 
-extern "C" int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size)
-{
-	char* text = new char[Size + 1];
-	memcpy(text, Data, Size);
-	text[Size] = 0;
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size) {
+  FuzzedDataProvider fdp(Data, Size);
+  std::string text = fdp.ConsumeRandomLengthString(1024);
 
-#ifdef PUGIXML_NO_EXCEPTIONS
-	pugi::xpath_query q(text);
-#else
-	try
-	{
-		pugi::xpath_query q(text);
-	}
-	catch (pugi::xpath_exception&)
-	{
-	}
-#endif
+  try {
+    pugi::xpath_variable_set vars;
+    size_t var_count = fdp.ConsumeIntegralInRange<size_t>(0, 50);
+    std::vector<std::string> var_name_storage = {};
+    for (size_t i = 0; i < var_count; i++) {
+      var_name_storage.push_back(fdp.ConsumeRandomLengthString(128));
+      vars.add(var_name_storage.back().c_str(), fdp.ConsumeEnum<pugi::xpath_value_type>());
```

Comment:
The `kMaxValue` (modification to `pugixml.hpp`) is required `ConsumeEnum` to work correctly.

Reply:
For posterity here is where `kMaxValue` is actually used by `FuzzedDataProvider::ConsumeEnum` https://github.com/llvm/llvm-project/blob/main/compiler-rt/include/fuzzer/FuzzedDataProvider.h#L298